cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapStatusEvaluator.java

```diff
@@ -89,12 +99,39 @@ private CareGapsStatusCode getGapStatus(Measure measure, MeasureReportGroupCompo
 
         if (Boolean.TRUE.equals(inDenominator.getValue())
                 && ((isPositive && !inNumerator.getValue()) || (!isPositive && inNumerator.getValue()))) {
-            return CareGapsStatusCode.OPEN_GAP;
+            return getOpenOrProspectiveStatus(measureReportGroup, reportDate);
         }
 
         return CareGapsStatusCode.CLOSED_GAP;
     }
 
+    private CareGapsStatusCode getOpenOrProspectiveStatus(
+            MeasureReportGroupComponent measureReportGroup, DateTimeType reportDate) {
+        if (hasDateOfComplianceExt(measureReportGroup)) {
+            Period dateOfCompliance = getDateOfComplianceExt(measureReportGroup);
+
+            boolean reportBeforeEndOfDOC = reportDate.before(dateOfCompliance.getEndElement());
+            boolean reportAfterStartOfDOC = reportDate.after(dateOfCompliance.getStartElement());
+            boolean reportBeforeStartOfDOC = reportDate.before(dateOfCompliance.getStartElement());
+
+            boolean reportWithinDOC = reportAfterStartOfDOC && reportBeforeEndOfDOC;
+            if (reportWithinDOC || reportBeforeStartOfDOC) {
+                return CareGapsStatusCode.PROSPECTIVE_GAP;
+            }
+        }
+        return CareGapsStatusCode.OPEN_GAP;
+    }
+
+    private boolean hasDateOfComplianceExt(MeasureReportGroupComponent measureReportGroup) {
+        var ext = measureReportGroup.getExtensionByUrl(CQFM_CARE_GAP_DATE_OF_COMPLIANCE_EXT_URL);
+        return ext != null && !ext.getValue().isEmpty();
+    }
+
+    private Period getDateOfComplianceExt(MeasureReportGroupComponent measureReportGroup) {
+        var ext = measureReportGroup.getExtensionByUrl(CQFM_CARE_GAP_DATE_OF_COMPLIANCE_EXT_URL);
+        return (Period) ext.getValue();
```

Comment:
Could this ever not be a Period?  Should we error handle for this?

Reply:
added assertion, this should always be a period